Formula/smap.rb

```diff
@@ -0,0 +1,23 @@
+class Smap < Formula
+  desc "Is a drop-in replacement for Nmap powered by shodan.io"
+  homepage "https://github.com/s0md3v/Smap"
+  url "https://github.com/s0md3v/Smap/archive/refs/tags/0.1.11.tar.gz"
+  sha256 "001088c3b530e3551a5014047c26e77953c096b39f0b1f874fb02d557552e07c"
+  license "AGPL-3.0-or-later"
+  head "https://github.com/s0md3v/Smap.git", branch: "main"
+
+  livecheck do
+    url :stable
+    regex(/^v?(\d+(?:\.\d+)+)$/i)
+  end
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", *std_go_args(ldflags: "-s -w"), "./cmd/..."
+  end
+
+  test do
+    assert_match "Usage: smap <targets here>", shell_output("#{bin}/smap")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
done this like nmap
https://github.com/Homebrew/homebrew-core/blob/6183f8d007af2c39df875fe119ce9aac1539febb/Formula/nmap.rb#L64

Reply:
Yeah, looks fine